### ISSUE TITLE
Feature: duplicate transaction detection for CSV imports

### ISSUE THREAD

**Issue by mgirouard** [maybe-finance/maybe#1214]
**Describe the bug**
After a few weeks of using Maybe, I noticed that I find myself deleting duplicate transactions often. I regularly export CSVs from my banks and import them. It's easier for me to grab the last 30 days of activity and import them without having to grab specific dates, but when I do this I always wind up with duplicates.

**To Reproduce**
1. Go to any account
2. Click on the ellipsis (`...`) menu and select Import transactions
3. Follow the normal import flow (adjust fields, adjust data, confirm transactions)
4. Navigate back to the account and review the latest transactions

(Perhaps a quicker repro would be to create a temporary account and import transactions twice)

**Expected behavior**
Duplicate transactions should be filtered out and not be imported.

**Screenshots / Recordings**
n/a

**Additional context**
Perhaps this could be done by hashing the transaction data -- exact matches would be a high probability of being duplicates (same description, same amount, same date, etc).

There will almost certainly be edge cases though -- eg: if I intentionally make the same purchase twice. One of the screens during import might be a good place to call this out and give me a checkbox to import it in or not.

**Comment by NGUYENDUCCHUNG21** [comment]
_This text is italicized_

**Comment by zachgoll** [comment]
@mgirouard currently, we do not have any support for detecting duplicates, but I agree that the feature would be useful.

I agree with you that hashing the raw transaction data is probably the best option for detecting matches.

What I'm not 100% sure about is the UI/UX for this (cc @justinfar)

My guess is that we could provide some sort of duplicate filtering / removal in our "clean" step?

![CleanShot 2024-09-30 at 10 17 35@2x](https://github.com/user-attachments/assets/e6a92f03-5be6-48fe-8c90-aa5315fee6c3)

**Comment by mgirouard** [comment]
@zachgoll Yeah, this makes the most sense to me. Thanks for considering :)

**Comment by NGUYENDUCCHUNG21** [comment]
Cái gì dậy ?

Vào Th 3, 1 thg 10, 2024 lúc 09:00 Michael Girouard <
***@***.***> đã viết:

**Comment by vanhalt** [comment]
I was thinking about this. But by hashing the transaction we may be discarding duplicated charges:

```csv
10/07/2024,Apple Store,-36
10/07/2024,Apple Store,-36
```

**Comment by zachgoll** [comment]
@vanhalt yeah, that's why I'm thinking we'll need a UI/UX that simply _surfaces_ duplicates to the user and then lets them decide whether to remove them or not.

**Comment by m3Lith** [comment]
This is how Actual does it:

![Image](https://github.com/user-attachments/assets/44c176c9-4701-4cf6-ad4e-5e111a7a46e2)

They have a toggleable icon on the left, for updating, importing again (duplicating), or ignoring. By default, identical transactions are ignored, and those with similar dates and identical amounts are updated.

**Comment by steebchen** [comment]
For me this would be quite important as I'm always importing with CSV, and there will be always duplicates unless I always set the exact correct dates when exporting CSVs from multiple banks, so this is quite error-prone and I will always end up with duplicates. So that makes it quite unusable for me unfortunately :/
